internal/services/nginx/nginx_deployment_data_source.go

```diff
@@ -261,12 +291,28 @@ func (m DeploymentDataSource) Read() sdk.ResourceFunc {
 					}
 
 					if scaling := props.ScalingProperties; scaling != nil {
-						output.Capacity = pointer.ToInt64(props.ScalingProperties.Capacity)
+						if capacity := scaling.Capacity; capacity != nil {
+							output.Capacity = pointer.ToInt64(props.ScalingProperties.Capacity)
```

Comment:
I'm a little confused by these types. I'm having a hard time navigating through the layers, but it looks like `props.ScalingProperties.Capacity` is an int64. That means it can never be nil, so our `!= nil` guard is always true. Do we need our guard to be `!= 0`, or drop the guard altogether?

Reply:
Once we have the generated SDK, I think this review will be easier?

```shell
type NginxDeploymentScalingProperties struct {
	AutoScaleSettings *NginxDeploymentScalingPropertiesAutoScaleSettings `json:"autoScaleSettings,omitempty"`
	Capacity          *int64                                             `json:"capacity,omitempty"`
}
```

Capacity is an integer pointer.

Reply:
Going to resolve this based on my answer above. @ryepup let me know if you want to chat more about it.